src/coreclr/src/jit/importer.cpp

```diff
@@ -4126,43 +4126,26 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
 
             case NI_System_Type_IsAssignableFrom:
             {
-                // Optimize patterns like:
-                //
-                //   typeof(TTo).IsAssignableFrom(typeof(TTFrom))
-                //   valueTypeVar.GetType().IsAssignableFrom(typeof(TTFrom))
-                //
-                // to true/false
                 GenTree* typeTo   = impStackTop(1).val;
                 GenTree* typeFrom = impStackTop(0).val;
 
-                if (typeTo->IsCall() && typeFrom->IsCall())
+                GenTreeIntCon* node = impTypeIsAssignable(typeTo, typeFrom);
+                if (node != nullptr)
```

Comment:
I suspect this check is not needed since `retNode` is nullptr anyway at this point.

Reply:
Simplified